x-pack/plugins/security_solution/public/detection_engine/rule_creation/components/esql_query_edit/validators/esql_query_validator_factory.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { QueryClient } from '@tanstack/react-query';
+import type { DatatableColumn } from '@kbn/expressions-plugin/common';
+import type { ESQLAstQueryExpression, ESQLCommandOption } from '@kbn/esql-ast';
+import { parse } from '@kbn/esql-ast';
+import { isAggregatingQuery } from '@kbn/securitysolution-utils';
+import { isColumnItem, isOptionItem } from '@kbn/esql-validation-autocomplete';
+import type { FormData, ValidationError, ValidationFunc } from '../../../../../shared_imports';
+import type { FieldValueQueryBar } from '../../../../rule_creation_ui/components/query_bar_field';
+import { fetchEsqlQueryColumns } from '../../../logic/esql_query_columns';
+import { ESQL_ERROR_CODES } from './error_codes';
+import * as i18n from './translations';
+
+interface EsqlQueryValidatorFactoryParams {
+  queryClient: QueryClient;
+  /**
+   * This is a temporal fix to unlock prebuilt rule customization workflow
+   */
+  skipIdColumnCheck?: boolean;
```

Comment:
Without id alerts won't be deduplicated while rule running and source fields might be missing in alert. This is a very important check
When is this planned to be addressed? Do we have a ticket?

Reply:
We need to have the Prebuilt Rule Customization workflow unlocked for testing. It's planned to address _id column validation right after implementing the most critical functionality to fully unlock the workflow.

Reply:
Good to hear it is planned to be addressed before release.
It was not obvious from a comment when it is going to be fixed and I was not part of that discussion